### PR TITLE
Solution?

### PR DESCRIPTION
**Pull request by Vivdyuk**
[DEMO LINK](https://youtu.be/dQw4w9WgXcQ)
 [DEMO LINK](https://vivdyuk.github.io/react_dynamic-list-of-goods/)

### CODE REVIEW

**Review by tanyakovchuk — APPROVED** [review]
🚀

File: src/App.js [hunk 1]
```diff
@@ -2,12 +2,56 @@ import React from 'react';
...
+    const goods = await get5First();
+
+    this.setState({ goods });
+  }
+
+  getAllRed = async() => {
+    const goods = await getRedGoods();
+
+    this.setState({ goods });
+  }
```
Comment (solaryasha) [inline src/App.js:30]: [OMG](https://www.youtube.com/watch?v=dQw4w9WgXcQ)
Почитай оцей тред - https://mateacademy.slack.com/archives/C021CDS2J79/p1627973315266000

**Review by solaryasha — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (Vivdyuk) [reply src/App.js]: [Якось так?](http://joxi.ru/YmE5OqlUBGDqLm)

File: src/api/goods.js [hunk 2]
```diff
@@ -6,6 +6,9 @@ export function getAll() {
     .then(response => response.json());
 }
 
-export const get5First = () => {};
+export const get5First = () => getAll()
+  .then(goods => goods
+    .sort((a, b) => a.name.localeCompare(b.name)).slice(0, 5));
```
Comment (solaryasha) [inline src/api/goods.js:11]:
```suggestion
    .sort((a, b) => a.name.localeCompare(b.name))
    .slice(0, 5));
```

**Review by solaryasha — APPROVED** [review]

File: src/App.js [hunk 3]
```diff
@@ -2,12 +2,44 @@ import React from 'react';
...
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+class App extends React.Component {
+  state = {
+    goods: [],
+  }
+
+ getGoods = async(method) => {
```
Comment (solaryasha) [inline src/App.js:14]:
```suggestion
 setGoods = async(getGoods) => {
```

**Pull request closed without merging**